### PR TITLE
replace null driver with local

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: code/iaas/service-discovery/api/src/main/java/io/cattle/platform/compose/export/impl/ComposeExportServiceImpl.java [hunk 1]
```diff
@@ -506,6 +507,9 @@ private void translateV1VolumesToV2(Map<String, Object> cattleServiceData,
         if (StringUtils.isEmpty(volumeDriver)) {
             return;
         }
+        if ("null".equals(volumeDriver)) {
```
Comment (cjellick) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/compose/export/impl/ComposeExportServiceImpl.java]: Please add a test. Looks like there's some existing tests to go off of:
```
cattle$ grep -Iir compose tests/integration/cattletest/ | grep expo
tests/integration/cattletest/core/test_svc_selectors.py:    compose_config = env.exportconfig()
tests/integration/cattletest/core/test_svc_volume_template.py:    compose_config = stack.exportconfig()
tests/integration/cattletest/core/test_svc_volume_template.py:    compose_config = stack.exportconfig()
tests/integration/cattletest/core/test_svc_volume_template.py:    compose_config = stack.exportconfig()
tests/integration/cattletest/core/test_svc_volume_template.py:    compose_config = stack.exportconfig()
tests/integration/cattletest/core/test_svc_discovery.py:    compose_config = env.exportconfig()
tests/integration/cattletest/core/test_svc_discovery.py:    compose_config = env.exportconfig()
tests/integration/cattletest/core/test_svc_discovery.py:    compose_config = env.exportconfig()
tests/integration/cattletest/core/test_svc_discovery.py:    compose_config = env.exportconfig()
tests/integration/cattletest/core/test_balancer_svc.py:    compose_config = env.exportconfig()
tests/integration/cattletest/core/test_balancer_svc.py:    compose_config = env.exportconfig()
tests/integration/cattletest/core/test_balancer_svc.py:    compose_config = env2.exportconfig()
tests/integration/cattletest/core/test_healthcheck.py:    # test rancher-compose export
tests/integration/cattletest/core/test_healthcheck.py:    compose_config = env.exportconfig()
```

Comment (cjellick) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/compose/export/impl/ComposeExportServiceImpl.java]: wrong fix. The correct fix is to not use String.valudOf() above. That is returning "null" when the value is null.

**Comment by cjellick** [comment]
Darn it, this doesnt round trip in the following scenario:
```
version: 2
volumes:
  test1:
    external: false
services:
  n1:
    image: nginx
    volumes:
    - test1:/zzz
```
because the exported config looks like this:
```
version: '2'
services:
  n1:
    image: nginx
    volumes:
    - test1:/zzz
```

**Comment by StrongMonkey** [comment]
```
if stack.isCreateFromAYamlFile() {
    give them original file;
} else{
    stack.exportStack()
}
```

**Comment by cjellick** [comment]
@StrongMonkey this needs a second test for the other use case we addressed of having a volume template defined like:
```
volumes:
  foo: {}
  bar:
    external: false
```

**Comment by cjellick** [comment]
LGTM

**Comment by StrongMonkey** [comment]
version: '2'
volume:
  test1: {}
services:
  n1:
    image: nginx
    volumes:
    - test1:/zzz

**Pull request merged**